### PR TITLE
🚧Requirements added to hotfix pr status check target

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

File: CICD/CICD.Requirements.cs [hunk 1]
```diff
@@ -215,33 +215,49 @@ private bool ThatPRHasLabels()
         return true;
     }
 
-    private bool ThatThePRHasTheLabel(string labelName)
+    /// <summary>
+    /// Returns a value indicating if the pull request contains a label that matches the given <paramref name="labels"/>.
```
Comment (KinsonDigitalAdmin) [inline CICD/CICD.Requirements.cs:219]: Refactor to below:

```suggestion
    /// Returns a value indicating whether or not the pull request contains a label that matches the given <paramref name="labels"/>.
```

File: CICD/CICD.StatusChecks.cs [hunk 2]
```diff
@@ -134,8 +147,8 @@ public partial class CICD // StatusChecks
             () => ThatAllMilestoneIssuesHaveLabels(),
             () => ThatAllOfTheReleaseMilestoneIssuesAreClosed(ReleaseType.Preview, true),
             () => ThatAllOfTheReleaseMilestonePullRequestsAreClosed(ReleaseType.Preview, true),
-            () => ThatTheReleaseMilestoneOnlyContainsSingle(ReleaseType.Preview, ItemType.Issue),
-            () => ThatTheReleaseMilestoneOnlyContainsSingle(ReleaseType.Preview, ItemType.PullRequest),
+            () => ThatTheReleaseMilestoneOnlyContainsSingleReleaseItem(ReleaseType.Preview, ItemType.Issue),
+            () => ThatTheReleaseMilestoneOnlyContainsSingleReleaseItem(ReleaseType.Preview, ItemType.PullRequest),
```
Comment (KinsonDigitalAdmin) [inline CICD/CICD.StatusChecks.cs:151]: Remove these requirement methods

File: CICD/CICD.StatusChecks.cs [hunk 3]
```diff
@@ -159,8 +172,8 @@ public partial class CICD // StatusChecks
             () => ThatAllMilestoneIssuesHaveLabels(),
             () => ThatAllOfTheReleaseMilestoneIssuesAreClosed(ReleaseType.Production, true),
             () => ThatAllOfTheReleaseMilestonePullRequestsAreClosed(ReleaseType.Production, true),
-            () => ThatTheReleaseMilestoneOnlyContainsSingle(ReleaseType.Production, ItemType.Issue),
-            () => ThatTheReleaseMilestoneOnlyContainsSingle(ReleaseType.Production, ItemType.PullRequest),
+            () => ThatTheReleaseMilestoneOnlyContainsSingleReleaseItem(ReleaseType.Production, ItemType.Issue),
+            () => ThatTheReleaseMilestoneOnlyContainsSingleReleaseItem(ReleaseType.Production, ItemType.PullRequest),
```
Comment (KinsonDigitalAdmin) [inline CICD/CICD.StatusChecks.cs:176]: Remove these requirement methods

File: CICD/ExtensionMethods.cs [hunk 4]
```diff
@@ -889,6 +889,27 @@ public static async Task<Milestone> CloseMilestone(this IMilestonesClient client
...
+
+    /// <summary>
+    /// Returns a value indicating whether or not the GitHub issue is a pull request.
+    /// </summary>
+    /// <param name="issue">The GitHub issue to check.</param>
+    /// <returns><c>true</c> if a standard issue.</returns>
+    public static bool IsPullRequest(this Issue issue) => issue.PullRequest is not null;
+
+    /// <summary>
+    /// Returns a value indicating if a GitHub issue is a release to do issue.
```
Comment (KinsonDigitalAdmin) [inline CICD/ExtensionMethods.cs:907]: Refactor code comments

```suggestion
    /// Returns a value indicating whether or not a GitHub issue is a release to do issue.
```

File: CICD/ExtensionMethods.cs [hunk 5]
```diff
@@ -899,13 +920,20 @@ public static bool IsReleaseToDoIssue(this Issue issue, ReleaseType releaseType)
...
-        var isIssue = issue.PullRequest is null;
+        var isIssue = issue.IsPullRequest();
         var validTitle = issue.Title == releaseLabelOrTitle;
         var validLabelType = issue.Labels.Any(l => l.Name == releaseLabelOrTitle);
 
         return isIssue && validTitle && validLabelType;
     }
 
+    /// <summary>
+    /// Returns a value indicating if a GitHub issue is a release pull request.
```
Comment (KinsonDigitalAdmin) [inline CICD/ExtensionMethods.cs:931]: Refactor code comments

```suggestion
    /// Returns a value indicating whether or not a GitHub issue is a release pull request.
```

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

File: CICD/CICD.Requirements.cs [hunk 6]
```diff
@@ -729,7 +815,17 @@ private bool ThatTheReleaseMilestoneContainsIssues()
         return false;
     }
 
-    private bool ThatTheReleaseMilestoneOnlyContainsSingle(ReleaseType releaseType, ItemType itemType)
+    /// <summary>
+    /// Returns a value indicating whether or not a milestone only contains a single release item
+    /// of the given <paramref name="releaseType"/> and <paramref name="itemType"/>.
+    /// </summary>
+    /// <param name="releaseType">The type of release of the item.</param>
```
Comment (kselena) [inline CICD/CICD.Requirements.cs:822]: Wordy
```suggestion
    /// <param name="releaseType">The type of item release.</param>
```

File: CICD/CICD.Requirements.cs [hunk 7]
```diff
@@ -729,7 +815,17 @@ private bool ThatTheReleaseMilestoneContainsIssues()
...
+    /// <param name="itemType">The type of item.</param>
+    /// <returns><c>true</c> if only single item exists in the milestone.</returns>
```
Comment (kselena) [inline CICD/CICD.Requirements.cs:824]: Grammar
```suggestion
    /// <returns><c>true</c> if only a single item exists in the milestone.</returns>
```

File: CICD/ExtensionMethods.cs [hunk 8]
```diff
@@ -889,6 +889,27 @@ public static async Task<Milestone> CloseMilestone(this IMilestonesClient client
         return updatedMilestone;
     }
 
+    /// <summary>
+    /// Returns a value indicating whether or not the GitHub issue is a standard issue.
+    /// </summary>
+    /// <param name="issue">The GitHub issue to check.</param>
+    /// <returns><c>true</c> if a standard issue.</returns>
```
Comment (kselena) [inline CICD/ExtensionMethods.cs:896]: Refactor to below, throughout the codebase.
```suggestion
    /// <returns><c>true</c> if the issue is standard.</returns>
```

File: CICD/ExtensionMethods.cs [hunk 9]
```diff
@@ -889,6 +889,27 @@ public static async Task<Milestone> CloseMilestone(this IMilestonesClient client
...
+    public static bool IsPullRequest(this Issue issue) => issue.PullRequest is not null;
+
+    /// <summary>
+    /// Returns a value indicating whether or not a GitHub issue is a release to do issue.
+    /// </summary>
+    /// <param name="issue">The GitHub issue to check.</param>
+    /// <param name="releaseType">The type of release issue.</param>
+    /// <returns><c>true</c> if a release pull request.</returns>
```
Comment (kselena) [inline CICD/ExtensionMethods.cs:911]: Refactor to below throughout the code base
```suggestion
    /// <returns><c>true</c> if it's a release pull request.</returns>
```

**Review by kselena — CHANGES_REQUESTED** [review]

**Review by kselena — DISMISSED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]
